src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -196,6 +204,32 @@ export class WorkspaceSavedObjectsClientWrapper {
     return hasPermission;
   }
 
+  // Data source is a workspace level object, validate if the request has access to the data source within the requested workspace.
+  private validateDataSourcePermissions = (
```

Comment:
@yubonluo Could you please check if this change is compatible with the bulkGet changes you made recently regarding permission check?

Reply:
Sure, the data source validation has been moved to id_consumer_wrapper, you can delete the method from the permission_wrapper.
related PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8888/files#diff-6f4daf9988f5262d2403821efd458faf7157c7e7473ec65a16ae2724a92d6556L209